# Import libraries

In [1]:
import numpy as np
import pandas as pd

# Import dataset

In [2]:
xls = pd.ExcelFile('data-set.xlsx')
df_rent = pd.read_excel(xls, 'SEWAATM')
df_act = pd.read_excel(xls, 'Data Aktifitas')

# Data cleaning on "SEWAATM" sheet

In [3]:
# eliminate the first three empty rows 
df_rent_drop_3_first_row = df_rent.drop([0,1,2], axis=0)
df_rent_drop_3_first_row

PT. XXXXXXXXXXXXXX           Unnamed: 1   Unnamed: 2      Unnamed: 3  \
3                     NO      Tanggal Request  Nama LOKASI            KOTA   
4                    NaN                  NaN          NaN             NaN   
5                    NaN                  NaN          NaN             NaN   
6                      1  2010-09-21 00:00:00     PLN Kota  Jkt-HUB II RPC   
7                      2  2010-09-21 00:00:00     Perumnas   Jkt-HUB I RPC   
...                  ...                  ...          ...             ...   
65775                NaN                  NaN          NaN             NaN   
65776                NaN                  NaN          NaN             NaN   
65777                NaN                  NaN          NaN             NaN   
65778                NaN                  NaN          NaN             NaN   
65779                NaN                  NaN          NaN             NaN   

      Unnamed: 4           Unnamed: 5 Unnamed: 6           Unnamed: 7  \
3      MASA SEWA             Delivery       Type          Jatuh tempo   
4        (TAHUN)                  NaN        NaN            Tag 3-4th   
5            NaN                  NaN        NaN                  NaN   
6            NaN  2010-10-07 00:00:00        522  2013-10-01 00:00:00   
7            NaN  2010-10-07 00:00:00        522  2013-10-01 00:00:00   
...          ...                  ...        ...                  ...   
65775        NaN                  NaN        NaN                  NaN   
65776        NaN                  NaN        NaN                  NaN   
65777        NaN                  NaN        NaN                  NaN   
65778        NaN                  NaN        NaN                  NaN   
65779        NaN  2016-08-12 00:00:00        NaN                  NaN   

      Unnamed: 8           Unnamed: 9  ... Unnamed: 269 Unnamed: 270  \
3        Tagihan                  NaN  ...          NaN      Tagihan   
4              I                  NaN  ...          NaN           99   
5       No. Inv.              Periode  ...         JT48     No. Inv.   
6      379-11-10  2010-10-01 00:00:00  ...          NaN          NaN   
7      379-11-10  2010-10-01 00:00:00  ...          NaN          NaN   
...          ...                  ...  ...          ...          ...   
65775        NaN                  NaN  ...          NaN          NaN   
65776        NaN                  NaN  ...          NaN          NaN   
65777        NaN                  NaN  ...          NaN          NaN   
65778        NaN                  NaN  ...          NaN          NaN   
65779        NaN                  NaN  ...          NaN          NaN   

      Unnamed: 271 Unnamed: 272 Unnamed: 273 Unnamed: 274 Unnamed: 275  \
3              NaN          NaN      Tagihan          NaN          NaN   
4              NaN          NaN          100          NaN          NaN   
5          Periode         JT48     No. Inv.      Periode         JT48   
6              NaN          NaN          NaN          NaN          NaN   
7              NaN          NaN          NaN          NaN          NaN   
...            ...          ...          ...          ...          ...   
65775          NaN          NaN          NaN          NaN          NaN   
65776          NaN          NaN          NaN          NaN          NaN   
65777          NaN          NaN          NaN          NaN          NaN   
65778          NaN          NaN          NaN          NaN          NaN   
65779          NaN          NaN          NaN          NaN          NaN   

      Unnamed: 276 Unnamed: 277 Unnamed: 278  
3          Tagihan          NaN          NaN  
4              101          NaN          NaN  
5         No. Inv.      Periode         JT48  
6              NaN          NaN          NaN  
7              NaN          NaN          NaN  
...            ...          ...          ...  
65775          NaN          NaN          NaN  
65776          NaN          NaN          NaN  
65777          

In [4]:
# Retrieve data from the first seven columns to be processed first
# the reason is because the first seven columns of data have a more organized data structure than the others,
# so that the processing is relatively easier

df1 = df_rent_drop_3_first_row.iloc[:,0:8]
df1

PT. XXXXXXXXXXXXXX           Unnamed: 1   Unnamed: 2      Unnamed: 3  \
3                     NO      Tanggal Request  Nama LOKASI            KOTA   
4                    NaN                  NaN          NaN             NaN   
5                    NaN                  NaN          NaN             NaN   
6                      1  2010-09-21 00:00:00     PLN Kota  Jkt-HUB II RPC   
7                      2  2010-09-21 00:00:00     Perumnas   Jkt-HUB I RPC   
...                  ...                  ...          ...             ...   
65775                NaN                  NaN          NaN             NaN   
65776                NaN                  NaN          NaN             NaN   
65777                NaN                  NaN          NaN             NaN   
65778                NaN                  NaN          NaN             NaN   
65779                NaN                  NaN          NaN             NaN   

      Unnamed: 4           Unnamed: 5 Unnamed: 6           Unnamed: 7  
3      MASA SEWA             Delivery       Type          Jatuh tempo  
4        (TAHUN)                  NaN        NaN            Tag 3-4th  
5            NaN                  NaN        NaN                  NaN  
6            NaN  2010-10-07 00:00:00        522  2013-10-01 00:00:00  
7            NaN  2010-10-07 00:00:00        522  2013-10-01 00:00:00  
...          ...                  ...        ...                  ...  
65775        NaN                  NaN        NaN                  NaN  
65776        NaN                  NaN        NaN                  NaN  
65777        NaN                  NaN        NaN                  NaN  
65778        NaN                  NaN        NaN                  NaN  
65779        NaN  2016-08-12 00:00:00        NaN                  NaN  

[65777 rows x 8 columns]

In [5]:
# Substitute new column names for the first 7 columns

kolom_baru = ['NO', 'TANGGAL REQUEST', 'NAMA LOKASI',
       'KOTA', 'MASA SEWA (TAHUN)', 'DELIVERY', 'TYPE',
       'JATUH TEMPO (TAG 3-4TH)']

In [6]:
df1.columns = kolom_baru

In [7]:
# Eliminate unused top 3 rows

df1 = df1.drop([3,4,5], axis=0)

In [8]:
# Perform processing in the next columns

df2 = df_rent_drop_3_first_row.iloc[:,8:]

In [9]:
# Based on my observations, JT-48 columns has no data at all and I decided
# to remove the column

df2 = df2.dropna(axis=1, thresh=10)
df2

Unnamed: 8           Unnamed: 9 Unnamed: 11          Unnamed: 12  \
3        Tagihan                  NaN     Tagihan                  NaN   
4              I                  NaN           2                  NaN   
5       No. Inv.              Periode    No. Inv.              Periode   
6      379-11-10  2010-10-01 00:00:00   379-11-10  2010-11-01 00:00:00   
7      379-11-10  2010-10-01 00:00:00   379-11-10  2010-11-01 00:00:00   
...          ...                  ...         ...                  ...   
65775        NaN                  NaN         NaN                  NaN   
65776        NaN                  NaN         NaN                  NaN   
65777        NaN                  NaN         NaN                  NaN   
65778        NaN                  NaN         NaN                  NaN   
65779        NaN                  NaN         NaN                  NaN   

      Unnamed: 14 Unnamed: 15 Unnamed: 17 Unnamed: 18 Unnamed: 20 Unnamed: 21  \
3         Tagihan         NaN     Tagihan         NaN     Tagihan         NaN   
4               3         NaN           4         NaN           5         NaN   
5        No. Inv.     Periode    No. Inv.     Periode    No. Inv.     Periode   
6       410-12-10         Des   011-01-11      Jan'11   054-02-11      Feb'11   
7       410-12-10         Des   011-01-11      Jan'11   054-02-11      Feb'11   
...           ...         ...         ...         ...         ...         ...   
65775         NaN         NaN         NaN         NaN         NaN         NaN   
65776         NaN         NaN         NaN         NaN         NaN         NaN   
65777         NaN         NaN         NaN         NaN         NaN         NaN   
65778         NaN         NaN         NaN         NaN         NaN         NaN   
65779         NaN         NaN         NaN         NaN         NaN         NaN   

       ... Unnamed: 261 Unnamed: 262 Unnamed: 264 Unnamed: 265 Unnamed: 267  \
3      ...      Tagihan          NaN      Tagihan          NaN      Tagihan   
4      ...           96          NaN           97          NaN           98   
5      ...     No. Inv.      Periode     No. Inv.      Periode     No. Inv.   
6      ...          NaN          NaN          NaN          NaN          NaN   
7      ...          NaN          NaN          NaN          NaN          NaN   
...    ...          ...          ...          ...          ...          ...   
65775  ...          NaN          NaN          NaN          NaN          NaN   
65776  ...          NaN          NaN          NaN          NaN          NaN   
65777  ...          NaN          NaN          NaN          NaN          NaN   
65778  ...          NaN          NaN          NaN          NaN          NaN   
65779  ...          NaN          NaN          NaN          NaN          NaN   

      Unnamed: 268 Unnamed: 270 Unnamed: 271 Unnamed: 273 Unnamed: 274  
3              NaN      Tagihan          NaN      Tagihan          NaN  
4              NaN           99          NaN          100          NaN  
5          Periode     No. Inv.      Periode     No. Inv.      Periode  
6              NaN          NaN          NaN          NaN          NaN  
7              NaN          NaN          NaN          NaN          NaN  
...            ...          ...          ...          ...          ...  
65775          NaN          NaN          NaN          NaN          NaN  
65776          NaN          NaN          NaN          NaN          NaN  
65777          NaN          NaN          NaN          NaN          NaN  
65778          NaN          NaN          NaN          NaN          NaN  
65779          NaN          NaN          NaN          NaN          NaN  

[65777 rows x 202 columns]

In [10]:
# for naming the new column no invoice and period, I looped the string by adding
# i-th statement

kolom_baru2 = []

for i in range(1,102) :
    
  kolom_baru2.extend((f"NO INV - TAGIHAN {i}",f"PERIODE - TAGIHAN {i}"))

In [11]:
# Substitute the new column name for the next columns
df2.columns = kolom_baru2

In [12]:
# Discard unused top 3 lines
df2 = df2.drop([3,4,5], axis=0)

In [13]:
# Merge dataframe 1 and dataframe 2 which have been separated previously
df3 = df1.join(df2)

In [14]:
# Identification of missing values ​​/ empty data in the dataset
df3.isnull().sum()

NO                       64522
TANGGAL REQUEST          64599
NAMA LOKASI              64523
KOTA                     64527
MASA SEWA (TAHUN)        65353
                         ...  
PERIODE - TAGIHAN 99     65749
NO INV - TAGIHAN 100     65752
PERIODE - TAGIHAN 100    65752
NO INV - TAGIHAN 101     65759
PERIODE - TAGIHAN 101    65759
Length: 210, dtype: int64

In [15]:
# total of missing values / data is empty

df3.isnull().sum().sum()

13721393

In [16]:
# Remove missing values ​​on datasets sorted by row and having at least 4 data

df3 = df3.dropna(axis=0, thresh=4)

In [17]:
df3.isnull().sum().sum()

170737

In [18]:
df3.describe()

NO      TANGGAL REQUEST          NAMA LOKASI      KOTA  \
count   1247                 1172                 1247      1247   
unique  1247                  260                 1167       258   
top        1  2020-01-20 00:00:00  Assalam Hypermarket  Surabaya   
freq       1                  106                    2        39   

        MASA SEWA (TAHUN)             DELIVERY       TYPE  \
count                 421                 1085       1141   
unique                  4                  448         18   
top                     4  2015-08-07 00:00:00  529EPP7CL   
freq                  353                   18        318   

       JATUH TEMPO (TAG 3-4TH) NO INV - TAGIHAN 1  PERIODE - TAGIHAN 1  ...  \
count                     1072                878                 1071  ...   
unique                     111                203                  114  ...   
top        2018-09-01 00:00:00          216-10-16  2015-09-01 00:00:00  ...   
freq                        48                 42                   48  ...   

       NO INV - TAGIHAN 97 PERIODE - TAGIHAN 97 NO INV - TAGIHAN 98  \
count                   59                   54                  39   
unique                   7                    6                   5   
top              123-05-19  2019-05-01 00:00:00           123-05-19   
freq                    15                   15                  14   

       PERIODE - TAGIHAN 98 NO INV - TAGIHAN 99 PERIODE - TAGIHAN 99  \
count                    39                  25                   25   
unique                    5                   4                    4   
top     2019-02-01 00:00:00           071-03-19  2019-03-01 00:00:00   
freq                     14                  14                   14   

       NO INV - TAGIHAN 100 PERIODE - TAGIHAN 100 NO INV - TAGIHAN 101  \
count                    22                    22                   15   
unique                    3                     3                    2   
top               096-04-19   2019-04-01 00:00:00            123-05-19   
freq                     14                    14                   14   

       PERIODE - TAGIHAN 101  
count                     15  
unique                     2  
top      2019-05-01 00:00:00  
freq                      14  

[4 rows x 210 columns]

BEFORE filtering missing values ​​>> 65774 rows × 210 columns

AFTER filtering missing values ​​>> 1247 rows × 210 columns

I assume this step is correct because the number of data rows after the missing values ​​filter is done = the number of data numbers and cities where these two data only have unique values

Percentage of filter missing values ​​~ 98%

In [19]:
#Change NO data to index

df3.set_index(['NO'], inplace=True)

In [20]:
# Final dataset for "SEWAATM" sheet

df3

TANGGAL REQUEST                 NAMA LOKASI              KOTA  \
NO                                                                        
1     2010-09-21 00:00:00                    PLN Kota    Jkt-HUB II RPC   
2     2010-09-21 00:00:00                    Perumnas     Jkt-HUB I RPC   
3     2010-09-21 00:00:00               Tip Top Depok  Jkt-HUB VIII RPC   
4     2010-09-21 00:00:00           Tip Top Pdk Bambu    Jkt-HUB VI RPC   
5     2010-09-21 00:00:00         Tip Top Rawamangun        JKT-HUB VII   
...                   ...                         ...               ...   
1243                  NaN  KK RS SARDJITO/ KAMPUS ITY        YOGYAKARTA   
1244                  NaN                   MARI MALL           MAKASAR   
1245                  NaN           BANDARA HASANUDIN           MAKASAR   
1246                  NaN              KK RS. SYAFIRA         PEKANBARU   
1247                  NaN            TASPEN PEKANBARU         PEKANBARU   

     MASA SEWA (TAHUN)             DELIVERY       TYPE  \
NO                                                       
1                  NaN  2010-10-07 00:00:00        522   
2                  NaN  2010-10-07 00:00:00        522   
3                  NaN  2010-10-07 00:00:00        522   
4                  NaN  2010-10-07 00:00:00        522   
5                  NaN  2010-10-07 00:00:00    529EPP7   
...                ...                  ...        ...   
1243                 4  2020-11-20 00:00:00  529EPP7CL   
1244                 4                  NaN     GRGW10   
1245                 4                  NaN     GRGW10   
1246                 4                  NaN     GRGW10   
1247                 4                  NaN     GRGW10   

     JATUH TEMPO (TAG 3-4TH) NO INV - TAGIHAN 1  PERIODE - TAGIHAN 1  \
NO                                                                     
1        2013-10-01 00:00:00          379-11-10  2010-10-01 00:00:00   
2        2013-10-01 00:00:00          379-11-10  2010-10-01 00:00:00   
3        2013-10-01 00:00:00          379-11-10  2010-10-01 00:00:00   
4        2013-10-01 00:00:00          379-11-10  2010-10-01 00:00:00   
5        2013-10-01 00:00:00          379-11-10  2010-10-01 00:00:00   
...                      ...                ...                  ...   
1243     2024-12-01 00:00:00                NaN  2020-12-01 00:00:00   
1244                     NaN                NaN                  NaN   
1245                     NaN                NaN                  NaN   
1246                     NaN                NaN                  NaN   
1247                     NaN                NaN                  NaN   

     NO INV - TAGIHAN 2  ... NO INV - TAGIHAN 97 PERIODE - TAGIHAN 97  \
NO                       ...                                            
1             379-11-10  ...                 NaN                  NaN   
2             379-11-10  ...                 NaN                  NaN   
3             379-11-10  ...                 NaN                  NaN   
4             379-11-10  ...                 NaN                  NaN   
5             379-11-10  ...           279-12-18  2018-12-01 00:00:00   
...                 ...  ...                 ...                  ...   
1243                NaN  ...                 NaN                  NaN   
1244                NaN  ...                 NaN                  NaN   
1245                NaN  ...                 NaN                  NaN   
1246                NaN  ...                 NaN                  NaN   
1247                NaN  ...                 NaN                  NaN   

     NO INV - TAGIHAN 98 PERIODE - TAGIHAN 98 NO INV - TAGIHAN 99  \
NO                                                                  
1                    NaN                  NaN                 NaN   
2                    NaN                  NaN                 NaN   
3                    NaN                  NaN                 NaN   
4                    NaN                  NaN  

# Data cleaning on "Data Aktifitas" sheet

In [21]:
df_act

Data aktivtas ATM        Unnamed: 1 Unnamed: 2    Unnamed: 3  \
0                  NaN               NaN        NaN           NaN   
1                  NaN               NaN        NaN           NaN   
2                  No.           Tanggal      Nomor        ATM ID   
3                    1      16 Juli 2020      88826      ATM00001   
4                    2     03 Maret 2020      86010  ATM120029008   
...                ...               ...        ...           ...   
3660              3168      01 Juli 2020      88730   ATM20099005   
3661              3169  09 November 2020      90731   ATM20099007   
3662              3170       27 Mei 2020      88655   ATM20099009   
3663               NaN               NaN        NaN           NaN   
3664              3171  22 Februari 2020      88642    ATM2013001   

                 Unnamed: 4     Unnamed: 5               Unnamed: 6  \
0                       NaN            NaN                      NaN   
1                       NaN            NaN                      NaN   
2                    Lokasi  Jenis Layanan        Laporan Kerusakan   
3            RUANG UAT DPTI         Error                 Hardware    
4            PLN AHMAD YANI         Error                 Hardware    
...                     ...            ...                      ...   
3660  BSB CAPEM KRAMAT JATI       Support                 Hardware    
3661    CAPEM BSD TANGERANG       Support                 Hardware    
3662   BSB  RS JATISAMPURNA    Error & PM             Non Hardware    
3663                    NaN            NaN                      NaN   
3664                    NaN          PM 1   Preventive Maintenance    

                                             Unnamed: 7   Unnamed: 8  \
0                                                   NaN          NaN   
1                                                   NaN          NaN   
2                                   Pekerjaan Dilakukan  Nomor Tiket   
3     Check & Clean all devices, replace ..............    2007-100    
4     Check & Clean all devices, replace EPP5 & CCA ...    2003-028    
...                                                 ...          ...   
3660  Check & Clean all devices, replace ....CRD.......    2007-094    
3661  Check & Clean all devices, replace ..............   2011-1432    
3662  Check & Clean all devices, replace ..............                
3663                                                NaN          NaN   
3664   Check & Clean all devices, no replace, test OK.                 

                                             Unnamed: 9  Unnamed: 10  \
0                                                   NaN          NaN   
1                                                   NaN          NaN   
2                                 Penggantian Sparepart  Jumlah Part   
3                                            Hard Disk            1    
4     CCA Dispenser (CCA Kit Presenter Ctrl.) 49-747...           1    
...                                                 ...          ...   
3660                             Card Reader Motorized            1    
3661                            Motherboard P 90 - Kit            1    
3662                                    Stripper Wheel            1    
3663                                        Feed Shaft            1    
3664                                                                   

     Unnamed: 11       Unnamed: 12 Unnamed: 13  
0            NaN               NaN         NaN  
1            NaN               NaN         NaN  
2         Status               CSE     Pilihan  
3                         M. Yani          NaN  
4                 Hendri Apridzal          NaN  
...          ...               ...         ...  
3660                      M. Yani          NaN  
3661                      M. Yani          NaN  
3662                      M. Yani          NaN  
3663                      M. Yani          NaN  
3664                 Hadi Warsito          NaN  

[366

In [22]:
# eliminate the first two lines because the two lines have no data / empty
df5 = df_act.drop([0,1], axis=0)

In [23]:
#Substitute new column names for the entire dataset

kolom_baru3 = ['NO', 'TANGGAL', 'NOMOR', 'ATM ID',
       'LOKASI', 'JENIS LAYANAN', 'LAPORAN KERUSAKAN', 'PEKERJAAN DILAKUKAN',
       'NOMOR TIKET', 'PENGGANTIAN SPAREPART', 'JUMLAH PART', 'STATUS',
       'CSE', 'PILIHAN']

In [24]:
df5.columns = kolom_baru3

In [25]:
# remove the top 1 row because that row has no data / blank / loop from column headings

df5.drop([2], axis=0, inplace = True)

In [26]:
# Based on my observations, some rows in the dataset have no data at all and I decided
# to remove the row with the criteria that there are at least 8 data in each row
# and eliminates rows that are less than the criteria

df6 = df5.dropna(axis=0, thresh=8)

In [27]:
# Remove "Pilihan" column because it has no value at all

df7 = df6.dropna(axis=1, how="all")

In [28]:
# Remove "Status" column because it has no value at all

df7 = df7.drop(['STATUS'], axis=1)

In [29]:
#Change NO data to index

df7.set_index(['NO'], inplace=True)

In [30]:
#Data "TANGGAL" in the dataset is a string and must be converted to a time format that can be read by the libraries used
#First stage: change the name of the month in the dataset because it is presented in Indonesian to the form
# the abbreviated name of the month that can be read by libraries

old_month = ["Januari", "Februari", "Maret", "April", "Mei", "Juni","Juli", "Agustus", "September", "Oktober", "November", "Desember"]
new_month = ["Jan", "Feb", "Mar", "Apr", "May", "Jun","Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]

In [31]:
df7['TANGGAL'] = df7['TANGGAL'].replace(old_month,new_month,regex=True)

In [32]:
#Convert data on "TANGGAL" in string format to timestamp format

df7["TANGGAL"] = pd.to_datetime(df7["TANGGAL"])
df7

TANGGAL  NOMOR        ATM ID                 LOKASI  JENIS LAYANAN  \
NO                                                                           
1    2020-07-16  88826      ATM00001         RUANG UAT DPTI         Error    
2    2020-03-03  86010  ATM120029008         PLN AHMAD YANI         Error    
3    2020-09-02  90681  ATM120044005           GRAHA ENERGY          PM 1    
4    2020-09-09  89466  ATM120047014                    NaN  Installation    
5    2020-03-29  87602   ATM20002001              BKP MEDAN     PM pertama   
...         ...    ...           ...                    ...            ...   
3167 2020-05-26  88731   ATM20099004      BSB CABANG BEKASI    Error & PM    
3168 2020-07-01  88730   ATM20099005  BSB CAPEM KRAMAT JATI       Support    
3169 2020-11-09  90731   ATM20099007    CAPEM BSD TANGERANG       Support    
3170 2020-05-27  88655   ATM20099009   BSB  RS JATISAMPURNA    Error & PM    
3171 2020-02-22  88642    ATM2013001                    NaN          PM 1    

            LAPORAN KERUSAKAN  \
NO                              
1                   Hardware    
2                   Hardware    
3     Preventive Maintenance    
4               Installation    
5     Preventive Maintenance    
...                       ...   
3167                Hardware    
3168                Hardware    
3169                Hardware    
3170            Non Hardware    
3171  Preventive Maintenance    

                                    PEKERJAAN DILAKUKAN NOMOR TIKET  \
NO                                                                    
1     Check & Clean all devices, replace ..............   2007-100    
2     Check & Clean all devices, replace EPP5 & CCA ...   2003-028    
3      Check & Clean all devices, no replace, test OK.                
4           Instalasi ATM, Check all devices, test OK.                
5      Check & Clean all devices, no replace, test OK.                
...                                                 ...         ...   
3167  Check & Clean all devices, replace ..............   2005-068    
3168  Check & Clean all devices, replace ....CRD.......   2007-094    
3169  Check & Clean all devices, replace ..............  2011-1432    
3170  Check & Clean all devices, replace ..............               
3171   Check & Clean all devices, no replace, test OK.                

                                  PENGGANTIAN SPAREPART JUMLAH PART  \
NO                                                                    
1                                            Hard Disk           1    
2     CCA Dispenser (CCA Kit Presenter Ctrl.) 49-747...          1    
3                                                                     
4                                                                     
5                                                                     
...                                                 ...         ...   
3167                            CA PWR, AC VGA 15 inch           1    
3168                             Card Reader Motorized           1    
3169                            Motherboard P 90 - Kit           1    
3170                                    Stripper Wheel           1    
3171                                                                  

                    CSE  
NO                       
1              M. Yani   
2      Hendri Apridzal   
3           ozi medina   
4              M. Yani   
5     Rajamin Nasution   
...                 ...  
3167           M. Yani   
3168           M. Yani   
3169           M. Yani   
3170           M. Yani   
3171      Hadi Warsito   

[3171 rows x 11 columns]

In [33]:
#Capitalize each letter in CSE data with the aim of uniformity

df7['CSE'] = df7['CSE'].str.upper()
df7

TANGGAL  NOMOR        ATM ID                 LOKASI  JENIS LAYANAN  \
NO                                                                           
1    2020-07-16  88826      ATM00001         RUANG UAT DPTI         Error    
2    2020-03-03  86010  ATM120029008         PLN AHMAD YANI         Error    
3    2020-09-02  90681  ATM120044005           GRAHA ENERGY          PM 1    
4    2020-09-09  89466  ATM120047014                    NaN  Installation    
5    2020-03-29  87602   ATM20002001              BKP MEDAN     PM pertama   
...         ...    ...           ...                    ...            ...   
3167 2020-05-26  88731   ATM20099004      BSB CABANG BEKASI    Error & PM    
3168 2020-07-01  88730   ATM20099005  BSB CAPEM KRAMAT JATI       Support    
3169 2020-11-09  90731   ATM20099007    CAPEM BSD TANGERANG       Support    
3170 2020-05-27  88655   ATM20099009   BSB  RS JATISAMPURNA    Error & PM    
3171 2020-02-22  88642    ATM2013001                    NaN          PM 1    

            LAPORAN KERUSAKAN  \
NO                              
1                   Hardware    
2                   Hardware    
3     Preventive Maintenance    
4               Installation    
5     Preventive Maintenance    
...                       ...   
3167                Hardware    
3168                Hardware    
3169                Hardware    
3170            Non Hardware    
3171  Preventive Maintenance    

                                    PEKERJAAN DILAKUKAN NOMOR TIKET  \
NO                                                                    
1     Check & Clean all devices, replace ..............   2007-100    
2     Check & Clean all devices, replace EPP5 & CCA ...   2003-028    
3      Check & Clean all devices, no replace, test OK.                
4           Instalasi ATM, Check all devices, test OK.                
5      Check & Clean all devices, no replace, test OK.                
...                                                 ...         ...   
3167  Check & Clean all devices, replace ..............   2005-068    
3168  Check & Clean all devices, replace ....CRD.......   2007-094    
3169  Check & Clean all devices, replace ..............  2011-1432    
3170  Check & Clean all devices, replace ..............               
3171   Check & Clean all devices, no replace, test OK.                

                                  PENGGANTIAN SPAREPART JUMLAH PART  \
NO                                                                    
1                                            Hard Disk           1    
2     CCA Dispenser (CCA Kit Presenter Ctrl.) 49-747...          1    
3                                                                     
4                                                                     
5                                                                     
...                                                 ...         ...   
3167                            CA PWR, AC VGA 15 inch           1    
3168                             Card Reader Motorized           1    
3169                            Motherboard P 90 - Kit           1    
3170                                    Stripper Wheel           1    
3171                                                                  

                    CSE  
NO                       
1              M. YANI   
2      HENDRI APRIDZAL   
3           OZI MEDINA   
4              M. YANI   
5     RAJAMIN NASUTION   
...                 ...  
3167           M. YANI   
3168           M. YANI   
3169           M. YANI   
3170           M. YANI   
3171      HADI WARSITO   

[3171 rows x 11 columns]

In [34]:
#Saving the processing results that have been applied by the cleaning process to a new excel file

with pd.ExcelWriter('Muhammad Fauzan Al-Baany_Dataset.xlsx') as writer:  # doctest: +SKIP
    df3.to_excel(writer, sheet_name='Sewa ATM')
    df7.to_excel(writer, sheet_name='Data Aktivitas')

print ("Export Telah Selesai")

Export Telah Selesai
